ee/modules/160-multitenancy-manager/webhooks/validating/projectType

```diff
@@ -0,0 +1,53 @@
+#!/usr/bin/env bash
+
+# Copyright 2023 Flant JSC
+# Licensed under the Deckhouse Platform Enterprise Edition (EE) license. See https://github.com/deckhouse/deckhouse/blob/main/ee/LICENSE
+
+source /shell_lib.sh
+
+function __config__(){
+  cat <<EOF
+configVersion: v1
+kubernetes:
+  - name: projects
+    apiVersion: deckhouse.io/v1alpha1
+    kind: Project
+    queue: "projects"
+    group: main
+    executeHookOnEvent: []
+    executeHookOnSynchronization: false
+    keepFullObjectsInMemory: false
+    jqFilter: |
+      {
+        "projectName": .metadata.name,
+        "projectTypeName": .spec.projectTypeName
+      }
+kubernetesValidating:
+- name: used-in-project.deckhouse.io
+  group: main
+  rules:
+  - apiGroups:   ["deckhouse.io"]
+    apiVersions: ["*"]
+    operations:  ["DELETE"]
+    resources:   ["projecttypes"]
+    scope:       "Cluster"
+EOF
+}
+
+function __main__() {
+  projectTypeName=$(context::jq -r '.review.request.object.spec.projectTypeName')
+
+  if usedInProjectName="$(context::jq -er --arg projectTypeName "$projectTypeName" '.snapshots.projects[].filterResult | select(.projectTypeName == $projectTypeName) | .name' 2>&1)"; then
+    cat <<EOF > "$VALIDATING_RESPONSE_PATH"
+{"allowed":false, "message":"projects.deckhouse.io \"$projectTypeName\", projectTypeName cannot be deleted. It is used in the project \"$usedInProjectName\"" }
+EOF
+    return 0
+  fi
+
+  cat <<EOF > "$VALIDATING_RESPONSE_PATH"
+{"allowed":true}
+EOF
+
+}
+
+hook::run "$@"
```

Comment:
Do you really prefer bash over python?

Reply:
It's the same for me.
Moreover, the hook next to it is also written in bash.